packages/trip-details/src/TripDetails.story.tsx

```diff
@@ -0,0 +1,251 @@
+
+const StyledTripDetails = styled(TripDetails)`
+  ${TripDetailsClasses.TripDetailsHeader} {
+    background-color: pink;
+  }
+`;
+
+const longDateFormat = "MMMM D, YYYY";
+
+// Custom slots for expandable detail sections.
+const CustomDepartureDetails = ({
+  departureDate
+}: DepartureDetailsProps): ReactElement => (
+  <>
+    Custom messages about {departureDate.format(longDateFormat)} can be
+    constructed dynamically using any markup.
+  </>
+);
+
+const CustomFareDetails = ({
+  maxTNCFare,
+  minTNCFare,
+  transitFare
+}: FareDetailsProps): ReactElement => (
+  <>
+    Custom details about fares (transitFare: {transitFare} (cents), minTNCFare:{" "}
+    {minTNCFare} and maxTNCFare: {maxTNCFare} can be constructed dynamically
+    using any markup.
+  </>
+);
+
+const CustomCaloriesDetails = ({
+  bikeSeconds,
+  calories,
+  walkSeconds
+}: CaloriesDetailsProps): ReactElement => (
+  <>
+    Custom message about {calories} calories burned,
+    {walkSeconds} seconds and {bikeSeconds} seconds.
+  </>
+);
+
+/**
+ * Create a template component for each TripDetails story.
+ * A Component argument is passed. Once bound,
+ * the Component to render should not change.
+ * @param Component The component to render.
+ */
+function createTripDetailsTemplate(Component = TripDetails) {
+  const TripDetailsTemplate = ({
+    CaloriesDetails,
+    currency,
+    DepartureDetails,
+    FareDetails,
+    itinerary
+  }: TripDetailsProps): React.Element => (
+    <Component
+      CaloriesDetails={CaloriesDetails}
+      currency={currency}
+      DepartureDetails={DepartureDetails}
+      FareDetails={FareDetails}
+      itinerary={itinerary}
+    />
+  );
+  return TripDetailsTemplate;
+}
+
+const intlDecorator = (
+  Story: StoryType,
+  context: {
+    locale?: string;
+    useCustomMessages?: boolean;
+    useLocalizedMessages?: boolean;
+  }
+): ReactElement => {
+  const { args } = context;
+  const { locale, useCustomMessages, useLocalizedMessages } = args;
+  const messages = flatten(
+    locale === "en-US" ? defaultEnglishMessages : defaultFrenchMessages
+  );
+  // Construct a messages object that customizes a subset
+  // of the default messages of the desired locale.
+  // The structure of the message objects is
+  // flattened before it is passed to IntlProvider.
+  const mergedMessages = useCustomMessages
+    ? {
+        ...messages,
+        ...flatten(customMessages)
+      }
+    : messages;
+
+  return (
+    <IntlProvider
+      locale={locale}
+      messages={useLocalizedMessages ? mergedMessages : null}
+    >
+      <Story />
+    </IntlProvider>
+  );
+};
+
+/**
+ * Helper to simplify story declaration.
+ */
+function makeStory(args, Component) {
+  const BoundTripDetails = createTripDetailsTemplate(Component).bind({});
+  BoundTripDetails.args = args;
+  return BoundTripDetails;
+}
```

Comment:
This might be what was causing all of those Storyshot snapshots to disappear. I think Storyshot might be unable to process dynamically generated stories like this.

Reply:
I spent some time replaying the commits of this PR, and a change in https://github.com/opentripplanner/otp-ui/pull/281/commits/6ef1c8493b2073386b830a10020dc237dd158ca6 is causing storyshot to throw out all of these snapshots. Digging further, it appears that importing the YML files is what is causing that, so we have to check that loading YML files is supported in storyshot.

Reply:
This one was really tricky, I have added a YML loader, tried to play well with other code, and updated the snapshots in https://github.com/opentripplanner/otp-ui/pull/281/commits/9725b4bf3d43303d383b1d42219fe199c3b7601e.